test/box-luatest/rollback_ddl_on__priv_space_test.lua

```diff
@@ -0,0 +1,51 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new()
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+-- Test that rollback of DDL statements on the `_priv` space works correctly.
+g.test_rollback_ddl_on__priv_space = function(cg)
+    cg.server:exec(function()
+        box.schema.user.create('grant')
+        box.schema.user.create('revoke')
+        box.schema.user.create('modify')
+
+        box.session.su('grant')
+        local grant_user = box.session.uid()
+        box.session.su('revoke')
+        local revoke_user = box.session.uid()
+        box.session.su('modify')
+        local modify_user = box.session.uid()
+
+        box.session.su('admin')
+
+        local grant_privs = box.space._priv:select{grant_user}
+        box.begin()
+        box.schema.user.grant('grant', 'execute', 'universe')
+        box.rollback()
+        t.assert_equals(box.space._priv:select{grant_user}, grant_privs)
```

Comment:
The test passes even without your patch (I reverted the _priv commit while being on top of your branch).

It seems you should check that actual operations are not allowed (output from bad tarantool version: note that `box.schema.user.info()` shows correct output, while the actual access is allowed) :
```lua
tarantool> box.schema.user.create('user')
---
...
tarantool> box.session.su('user')
---
...
tarantool> box.space._space:fselect{}
---
- error: Read access to space '_space' is denied for user 'user'
...
tarantool> box.session.su('admin')
---
...
tarantool> box.begin() box.schema.user.grant('user', 'read', 'space', '_space') box.rollback()
---
...
-- Note that the rolled back grant isn't present in `user.info()`, but actually still applies.
tarantool> box.schema.user.info('user')
---
- - - execute
    - role
    - public
  - - session,usage
    - universe
    - 
  - - alter
    - user
    - user
...
tarantool> box.session.su('user')
---
...
tarantool> box.space._space:fselect{}
---
- |-
  +-----+-----+------------------------+-----------+-----------+------------------+-----------------------------------------------+
  | id  |owner|          name          |  engine   |field_count|      flags       |                    format                     |
  +-----+-----+------------------------+-----------+-----------+------------------+-----------------------------------------------+
  | 257 |  1  |"_vinyl_deferred_delete"|"blackhole"|     0     |  {"group_id":1}  |[{"name":"space_id","type":"unsigned"},{"name":|
    
    <truncated for simplicity>

  | 380 |  1  |  "_session_settings"   | "service" |     2     |{"temporary":true}|[{"name":"name","type":"string"},{"name":"value|
  +-----+-----+------------------------+-----------+-----------+------------------+-----------------------------------------------+
...
```

Reply:
Fixed: tested the `grant`, `revoke`, `modify` of `read` and `write` privileges with operations on a `test` space.

Reply:
Thanks!